lib/Sema/ConstraintSystem.cpp

```diff
@@ -1509,16 +1488,35 @@ ConstraintSystem::getTypeOfMemberReference(
   }
 
   // Compute the type of the reference.
-  Type type;
+  Type type = openedType;
+
+  if (!outerDC->getSelfProtocolDecl()) {
```

Comment:
This is like the third or fourth time this block appears.  Can we centralize this?  One of them is going to fall out of sync one day...

Reply:
I only found two, but I'll extract it into a new method on ValueDecl.